### PR TITLE
PLNSRVCE-1153: Refactor service test pipeline run

### PR DESCRIPTION
**Pull request by prietyc123**
Simplify the pipeline Run for the tests and remove duplicates

Signed-off-by: Priti Kumari [pkumari@redhat.com](mailto:pkumari@redhat.com)

### CODE REVIEW

File: .tekton/pipeline-service-test.yaml [hunk 1]
```diff
@@ -29,27 +36,15 @@ spec:
...
-                CLUSTER_NAME="ci-$( openssl rand -hex 5 )"
-                echo -n "$CLUSTER_NAME" | tee $(results.cluster-name.path)
-      - name: clean-clusters
+      - name: generate-cluster-name
+        taskRef:
+          name: generate-cluster-name
+      - name: destroy-existing-cluster
+        taskRef:
+          name: destroy-existing-cluster
+          kind: ClusterTask
```
Comment (VeereshAradhya) [inline .tekton/pipeline-service-test.yaml:45]: remove this line

File: .tekton/pipeline-service-test.yaml [hunk 2]
```diff
@@ -58,46 +53,20 @@ spec:
...
-                  valueFrom:
-                    secretKeyRef:
-                      name: hypershift-bitwarden
-                      key: "BW_PASSWORD"
-              command:
-                - $(workspaces.source.path)/ci/images/ci-runner/hack/bin/destroy-clusters.sh
       - name: deploy-cluster
+        taskRef:
+          name: deploy-cluster
+          kind: ClusterTask
```
Comment (VeereshAradhya) [inline .tekton/pipeline-service-test.yaml:59]: remove this line

File: .tekton/pipeline-service-test.yaml [hunk 3]
```diff
@@ -6,7 +6,14 @@ metadata:
...
     pipelinesascode.tekton.dev/on-target-branch: "[refs/heads/*]"
-    pipelinesascode.tekton.dev/task: "[git-clone]"
+    pipelinesascode.tekton.dev/task: "[.tekton/tasks/ci-runner-container-setup.yaml]"
+    pipelinesascode.tekton.dev/task-1: "[.tekton/tasks/deploy-cluster.yaml]"
+    pipelinesascode.tekton.dev/task-2: "[.tekton/tasks/destroy-cluster.yaml]"
+    pipelinesascode.tekton.dev/task-3: "[.tekton/tasks/destroy-existing-cluster.yaml]"
+    pipelinesascode.tekton.dev/task-4: "[.tekton/tasks/generate-cluster-name.yaml]"
+    pipelinesascode.tekton.dev/task-5: "[.tekton/tasks/pipeline-service-setup.yaml]"
+    pipelinesascode.tekton.dev/task-6: "[.tekton/tasks/pipeline-service-tests.yaml]"
+    pipelinesascode.tekton.dev/task-7: "[git-clone]"
```
Comment (VeereshAradhya) [inline .tekton/pipeline-service-test.yaml:16]: if you are using `git-clone` clustertask, this line is not required

File: .tekton/pipeline-service-test.yaml [hunk 4]
```diff
@@ -166,7 +88,10 @@ spec:
             value: $(params.repo_url)
           - name: revision
             value: $(params.revision)
-      - name: setup-ci-runner-container
+      - name: setup-ci-runner
+        taskRef:
+          name: setup-ci-runner-container
+          kind: ClusterTask
```
Comment (VeereshAradhya) [inline .tekton/pipeline-service-test.yaml:94]: Remove this line

File: .tekton/pipeline-service-test.yaml [hunk 5]
```diff
@@ -177,40 +102,12 @@ spec:
...
-              workingDir: "$(workspaces.source.path)"
-              env:
-                - name: KUBECONFIG
-                  value: "$(workspaces.kubeconfig-dir.path)/kubeconfig"
-              command:
-                - $(workspaces.source.path)/ci/images/ci-runner/hack/bin/copy-plnsvc-code.sh
       - name: plnsvc-setup
+        taskRef:
+          name: pipeline-service-setup
+          kind: ClusterTask
```
Comment (VeereshAradhya) [inline .tekton/pipeline-service-test.yaml:108]: Remove this line

File: .tekton/pipeline-service-test.yaml [hunk 6]
```diff
@@ -223,33 +120,10 @@ spec:
...
-            - name: kubeconfig-dir
-            - name: source
-          params:
-            - name: repo_url
-            - name: revision
-            - name: target_branch
+      - name: pipeline-service-tests
+        taskRef:
+          name: pipeline-service-tests
           kind: ClusterTask
```
Comment (VeereshAradhya) [inline .tekton/pipeline-service-test.yaml:126]: Remove this line

File: .tekton/pipeline-service-test.yaml [hunk 7]
```diff
@@ -260,75 +134,25 @@ spec:
...
-              env:
-                - name: KUBECONFIG
-                  value: "$(workspaces.kubeconfig-dir.path)/kubeconfig"
-              command:
-                - $(workspaces.source.path)/ci/images/ci-runner/hack/bin/run-tests.sh
     finally:
       - name: destroy-cluster
+        taskRef:
+          name: destroy-cluster
+          kind: ClusterTask
```
Comment (VeereshAradhya) [inline .tekton/pipeline-service-test.yaml:141]: remove this line

File: .tekton/tasks/ci-runner-container-setup.yaml [hunk 8]
```diff
@@ -0,0 +1,35 @@
+apiVersion: tekton.dev/v1beta1
+kind: ClusterTask
```
Comment (VeereshAradhya) [inline .tekton/tasks/ci-runner-container-setup.yaml:2]: change this to `kind: Task`

File: .tekton/tasks/deploy-cluster.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
+apiVersion: tekton.dev/v1beta1
+kind: ClusterTask
```
Comment (VeereshAradhya) [inline .tekton/tasks/deploy-cluster.yaml:2]: Replace `ClusterTask` with `Task`

File: .tekton/tasks/destroy-cluster.yaml [hunk 10]
```diff
@@ -0,0 +1,36 @@
+apiVersion: tekton.dev/v1beta1
+kind: ClusterTask
```
Comment (VeereshAradhya) [inline .tekton/tasks/destroy-cluster.yaml:2]: Replace `ClusterTask` with `Task`

File: .tekton/tasks/destroy-existing-cluster.yaml [hunk 11]
```diff
@@ -0,0 +1,33 @@
+apiVersion: tekton.dev/v1beta1
+kind: ClusterTask
```
Comment (VeereshAradhya) [inline .tekton/tasks/destroy-existing-cluster.yaml:2]: Replace `ClusterTask` with `Task`

File: .tekton/tasks/pipeline-service-setup.yaml [hunk 12]
```diff
@@ -0,0 +1,31 @@
+apiVersion: tekton.dev/v1beta1
+kind: ClusterTask
```
Comment (VeereshAradhya) [inline .tekton/tasks/pipeline-service-setup.yaml:2]: Replace `ClusterTask` with `Task`

File: .tekton/tasks/pipeline-service-tests.yaml [hunk 13]
```diff
@@ -0,0 +1,23 @@
+apiVersion: tekton.dev/v1beta1
+kind: ClusterTask
```
Comment (VeereshAradhya) [inline .tekton/tasks/pipeline-service-tests.yaml:2]: Replace `ClusterTask` with `Task`

File: .tekton/pipeline-service-test.yaml (hunk 3 above)
Comment (VeereshAradhya) [reply .tekton/pipeline-service-test.yaml:16]: This downloads the task from tekton hub

**Comment by prietyc123** [comment]
Service test pipelineRun
image

**Comment by prietyc123** [comment]
> task plnsvc-setup has the status "Failed":
secret/signing-secrets created
error: accumulating resources: accumulation err='accumulating resources from 'tekton-results-secret.yaml': evalsymlink failure on '/source/developer/openshift/work/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml' : lstat /source/developer/openshift/work/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml: no such file or directory': must build at directory: not a valid directory: evalsymlink failure on '/source/developer/openshift/work/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml' : lstat /source/developer/openshift/work/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml: no such file or directory
command terminated with exit code 1

/retest pipeline-service-upgrade-test

**Pull request merged**